docs/ts/develop/debug.md

```diff
@@ -0,0 +1,56 @@
+---
+seotitle: How to debug your TS backend application
+seodesc: Learn how to debug your TS backend application using Encore.
+title: Debug with your IDE
+lang: ts
+---
+
+Encore makes it easy to debug your application using your favorite IDE. 
+
+## Enable debugging mode
+Next, run your Encore application with `encore run --debug`. This will cause Encore to run your app with the `--inspect-brk` flag, which will pause your application until a debugger is attached. Encore will print the URL to the terminal, which you will use to attach your debugger:
+
+```shell
+$ encore run --debug
+  Your API is running at:     http://127.0.0.1:4000
+  Development Dashboard URL:  http://localhost:9400/ai-chat-ts-qhwi
+  Process ID:                 38965
+
+Debugger listening on ws://127.0.0.1:9229/473dd95f-e71e-4bf2-9eda-6132dd0d6ae3
+```
+
+(Your process id and url will differ).
+
+## Attach your debugger
+When your Encore application is running, it’s time to attach the debugger. The instructions differ depending on how you would like to debug. If instructions for your editor aren’t listed below, consult your editor for information on how to attach a debugger to a running process.
+
+### Visual Studio Code
+To debug with VS Code you must first add a debug configuration. Press `Run -> Add Configuration`, choose `Node.js -> Attach`. The generated config should look something like this:
+
+```json
+{
+  "version": "0.2.0",
+  "configurations": [
+    {
+      "name": "Attach",
+      "port": 9229,
```

Comment:
Is this always 9229 or will it vary?

Reply:
it's 9229 by default. You can override it, but I haven't added support for it :). If 9229 isn't available, it will not init the debugger. If we want to make this more fail-safe we could either automatically look for an available port, or add an override debug port option.